src/Lean/PrettyPrinter/Delaborator/Basic.lean

```diff
@@ -255,19 +269,39 @@ an expression, which prevents terms such as atomic expressions or `OfNat.ofNat`
 delaborated as `⋯`.
 -/
 def shouldOmitExpr (e : Expr) : DelabM Bool := do
-  if ← getPPOption getPPDeepTerms then
+  -- Atomic expressions never get omitted, so we can do an early return here.
+  if e.isAtomic then
+    return false
+
+  if (← getPPOption getPPDeepTerms) then
     return false
 
   let depth := (← read).depth
   let depthThreshold ← getPPOption getPPDeepTermsThreshold
-  let approxDepth := e.approxDepth.toNat
   let depthExcess := depth - depthThreshold
+  -- This threshold for shallow expressions effectively allows full terms to be pretty printed 25% deeper,
+  -- so long as the subterm actually can be fully pretty printed within this extra 25% depth.
+  let threshold := depthThreshold/4 - depthExcess
 
-  let isMaxedOutApproxDepth := approxDepth >= 255
-  let isShallowExpression :=
-    !isMaxedOutApproxDepth && approxDepth <= depthThreshold/4 - depthExcess
+  return depthExcess > 0 && !isShallowExpression threshold e
 
-  return depthExcess > 0 && !isShallowExpression
+/--
+Returns `true` if the given expression is a proof and it should be omitted.
+This function only returns `true` if `pp.proofs` is set to `false`.
+It uses `pp.proofs.threshold` to determine what is a "shallow" proof -- shallow proofs are not omitted.
```

Comment:
"It uses `pp.proofs.threshold` to determine "shallow" proofs that are not omitted"?